functions/src/application/portal.ts

```diff
@@ -302,3 +303,31 @@ export const get_developer_profile = async (request: Request, response: Response
     });
   }
 };
+
+export const create_vanity_link = async (req: Request, res: Response) => {
+  const { first_name, last_name, email, destination, primary_domain, subdomain, slashtag } = req.body;
```

Comment:
Express supports generics for request types, so you can make this more strongly-typed by modifying line 307 to use an explicit request type. See this StackOverflow [answer](https://stackoverflow.com/a/55413670/5742625) for more info.